src/mongoose_rabbit_worker.erl

```diff
@@ -128,36 +133,55 @@ handle_amqp_publish(Method, Payload, Opts = #state{host = Host}) ->
     case Result of
         true ->
             update_messages_published_metrics(Host, PublishTime, Payload),
-            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]);
+            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]),
+            {noreply, Opts};
         false ->
             update_messages_failed_metrics(Host),
             ?WARNING_MSG("event=rabbit_message_sent_failed reason=negative_ack",
-                         []);
+                         []),
+            {noreply, Opts};
+        channel_exception ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=channel_exception",
+                         []),
+            {FreshConn, FreshChann} = restart_rabbit_connection(Opts),
+            {noreply, Opts#state{connection = FreshConn, channel = FreshChann}};
         timeout ->
             update_messages_timeout_metrics(Host),
             ?WARNING_MSG("event=rabbit_message_sent_failed reason=timeout",
-                         [])
+                         []),
+            {noreply, Opts}
     end.
 
 -spec publish_message_and_wait_for_confirm(Method :: mongoose_amqp:method(),
                                            Payload :: mongoose_amqp:message(),
                                            worker_opts()) ->
-    true | no_return().
+    boolean() | timeout | channel_exception.
 publish_message_and_wait_for_confirm(Method, Payload,
                                      #state{channel = Channel,
                                             confirms = IsConfirmEnabled}) ->
-    amqp_channel:call(Channel, Method, Payload),
-    case IsConfirmEnabled of
-        true ->
-            amqp_channel:wait_for_confirms(Channel);
-        false ->
-            true
+    case try_to_apply(amqp_channel, call, [Channel, Method, Payload]) of
+        {ok, _} ->
+            maybe_wait_for_confirms(Channel, IsConfirmEnabled);
+        {error, _, _} -> channel_exception
```

Comment:
I do not like idea of ignoring an error - we are not returning error or logging it.